### PR TITLE
[WIP] `cupy.around` behaves differently from NumPy for EVEN_NUMBER+0.5

### PR DESCRIPTION
**Pull request by hyabe**
`cupy.around` is incompatible with `numpy.around` such as `cupy.around(2.5) == numpy.around(2.5)` gives `False` (tested on CuPy 7.0.0b1 vs. NumPy 1.16.4).
This PR fixes the problem, and also solves #2276.

We may need additional fix and/or discussion because there still be incompatible case(s) even with this PR:
 * `cupy.around(2**63, -1)` differs from NumPy, the latter does *not* round the last digit,
 * corner cases like `cupy.array(255, cupy.uint8).round(-1)` are not exhaustively tested.

### CODE REVIEW

**Comment by grlee77** [comment]
Maybe the mixture of two separate changes in behavior is holding up review of this PR?

Would it make sense to split out just the switch to using `rint` in `_round_float` and `_round_complex` into a separate PR? I think that one definitely helps improve compatibility with NumPy and merging it should be non-controversial. 

The change in behavior for large integers implemented here looks desirable, but as noted, is not consistent with NumPy. Have you opened an issue (or is there a pre-existing one) over there?

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -1726,10 +1726,20 @@ if (in1 == 0) {
         inv_x = y;
     }
 }
-out0 = in0_type(round(in0.real() * x) * inv_x,
-                round(in0.imag() * x) * inv_x);'''
-
-
+out0 = in0_type(rint(in0.real() * x) * inv_x,
+                rint(in0.imag() * x) * inv_x);'''
```
Comment (grlee77) [inline cupy/core/core.pyx]: Analagous to the code in the real-valued case, is there a performance benefit to having just:
```C++
out0 = in0_type(rint(in0.real()), rint(in0.imag())
```
within the `in1 == 0` branch instead?

**Comment by take-cheeze** [comment]
ChainerCI, test this please!

File: tests/cupy_tests/core_tests/test_ndarray_math.py [hunk 2]
```diff
@@ -32,3 +32,84 @@ def test_round_out(self, xp):
...
+        a *= 2
+        a -= a.size + 1
+        scale = 10**abs(self.decimals)
+        if self.decimals < 0:
+            a *= scale
+        else:
+            a /= scale
+        a /= 2
+
+        print(xp, dtype, self.decimals, a, a.round(self.decimals))
```
Comment (take-cheeze) [inline tests/cupy_tests/core_tests/test_ndarray_math.py]: This print should be removed or commented out.

**Comment by take-cheeze** [comment]
> Would it make sense to split out just the switch to using rint in _round_float and _round_complex into a separate PR? I think that one definitely helps improve compatibility with NumPy and merging it should be non-controversial.

👍

**Comment by kmaehashi** [comment]
@hyabe Kindly ping.

**Comment by grlee77** [comment]
I ran into this issue again yesterday when looking at adding `mode='median'` support to `cupy.pad`.

If there is no objection, perhaps someone can take over and help complete the PR?

**Comment by kmaehashi** [comment]
Could you takeover? @emcastillo

**Comment by emcastillo** [comment]
Take over  in #3335 
Closing this

**Pull request closed without merging**